engine/classes/Elgg/Page/AddRssLinkHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Elgg\Page;
+
+/**
+ * Add RSS link to page head
+ *
+ * @since 4.0
+ */
+class AddRssLinkHandler {
+	
+	/**
+	 * Add rss link to HTML head
+	 *
+	 * @param \Elgg\Hook $hook 'head', 'page'
+	 *
+	 * @return array|void
+	 */
+	public function __invoke(\Elgg\Hook $hook) {
+		if (!_elgg_has_rss_link()) {
+			return;
+		}
+		
+		$head_params = $hook->getValue();
+		
+		$url = current_page_url();
+		if (elgg_substr_count($url, '?')) {
+			$url .= "&view=rss";
+		} else {
+			$url .= "?view=rss";
+		}
```

Comment:
why not use `elgg_http_add_url_query_elements()`
that would also override an already existing `view` in the query elements instead of adding a duplicate